lib/verify-service-provider-client.ts

```diff
@@ -17,9 +17,14 @@ export default class VerifyServiceProviderClient {
 
   constructor (private verifyServiceProviderHost: string) {}
 
-  async generateAuthnRequest (): Promise<{ status: number, body: AuthnRequestResponse | ErrorMessage }> {
+  async generateAuthnRequest (entityId?: string): Promise<{ status: number, body: AuthnRequestResponse | ErrorMessage }> {
     try {
-      const responseBody = await this.sendRequest<AuthnRequestResponse>('/generate-request', { levelOfAssurance: 'LEVEL_2' })
+      let requestBody: any = { levelOfAssurance: 'LEVEL_2' }
+      if (entityId) {
+        requestBody.entityId = entityId
+      }
```

Comment:
It would be shorter, and perhaps cleaner, to just pass entityId into the requestBody (which will then have the value of undefined if there is no entityId). But it's nicer from the point of view of the request to not pass anything if we don't have it, rather than passing undefined.